source/clear-linux/guides/maintenance/kernel-development.rst

```diff
@@ -0,0 +1,462 @@
+After the kernel sources have been obtained, customizations to the kernel 
+configuration or source code can be made for inclusion with the kernel 
+build.
+
+These customizations are option optional.
+
+
+Modify kernel configuration 
+===========================
+
+The kernel source has many configuration options available to pick 
+support for different hardware and software features.
+
+These configuration values must be provided in the :file:`.config` file at 
+compile time.
+
+You will need to make modifications to the :file:`.config` file, and include 
+it in the kernel package. 
+
+
+#. Make sure you have followed the steps to 
+   `Pull a copy of the Linux kernel source code`_ and are in the kernel 
+   source working directory.
+
+
+#. If you have an existing :file:`.config` file from an old kernel, copy it 
+   into the working directory as :file:`.config` for comparison.
+   Otherwise, generate a base kernel configuration. 
+
+   .. code-block:: bash
+
+      make defconfig
+
+   This will create a  :file:`.config` file within the working directory 
+   containing default values for the linux version.
+
+
+#. Make any desired changes to the :file:`.config` using a kernel 
+   configuration tool. Below are some popular options:
+
+   - :command:`$EDITOR .config` - the .config file can be directly edited for 
+     simple changes with names that are already known.
+
+   - :command:`make config` - a text-based tool that asks questions 
+     one-by-one to decide configuration options. 
+
+   - :command:`make menuconfig` - a terminal user interface that provides 
+     menus to decide configuration options.
+
+   - :command:`make xconfig` - a graphical user interface that provides 
+     tree views to decide configuration options.
+
+
+   More configuration tools can be found by looking at the make help: 
+   :command:`make help | grep config`
+
+
+#. Commit and save the changes to the :file:`.config` file.
+
+
+#. Copy the :file:`.config` file from the kernel source directory into 
+   the kernel package directory as :file:`config` for inclusion in the build.
+
+   .. code-block:: bash
+
+      cp .config ../config
+
+
+
+Modify kernel source code 
+=========================
+
+Changes to kernel code are applied with patch files. Patch files are 
+formatted git commits that can be applied to the main source code.
+
+You will need to obtain a copy of the source code,
+make modifications, generate patch file(s), and add them to the RPM SPEC 
+file for inclusion during the kernel build.
+
+If you have a large number of patches or a more complex workflow, 
+consider using a patch management tool in addition to Git such as 
+`Quilt`_. 
+
+
+#. Make sure you have followed the steps to 
+   `Pull a copy of the Linux kernel source code`_ and are in the kernel 
+   source working directory.
+
+
+#. Make any desired code changes to the Linux source code files.
+
+
+#. Track and commit your changes to the local git repo. 
+
+   .. code-block:: bash
```

Comment:
As the current instruction the following command will fail as it is not a git repository.

Reply:
It did not fail for me when I did it. It seems to do a `git init` on the local directory internally. It adds all files as new, and then we tell them only to stage/add the only the changed files

Did you try it and run into issue or can you recommend a better workflow that is both easy to read and execute?

Reply:
@miguelinux in any case, I have added an explicit `git init` and initial `git add -A` command to be explicit. Let me know if you think this is a good approach.

Reply:
:+1: